Providers/Profile.swift

```diff
@@ -135,7 +135,7 @@ class BrowserProfileSyncDelegate: SyncDelegate {
 /**
  * A Profile manages access to the user's data.
  */
-protocol Profile: class {
+protocol Profile: class, FirefoxAccountDelegate {
```

Comment:
Also, prefer to scope the name down -- `FirefoxAccountDeviceRegistrationDelegate` or whatever.  There may be more than one type of notification, right?